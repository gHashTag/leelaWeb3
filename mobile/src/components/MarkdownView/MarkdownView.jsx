/* eslint-disable react-native/no-unused-styles */
import React from 'react'

import { Platform, ScrollView, StyleSheet } from 'react-native'
import { View } from 'react-native'

import { NeomorphView } from 'components'
import { gray, W } from 'cons'
import Markdown from 'react-native-markdown-display'
import { ms, s } from 'react-native-size-matters'

import { Space } from '../Space/Space'

const MarkdownView = ({ markdown, children }) => {
  return (
    <NeomorphView>
      <ScrollView
        showsVerticalScrollIndicator={false}
        contentInsetAdjustmentBehavior="automatic"
        style={styles.container}
      >
        <Markdown style={styles}>{markdown}</Markdown>
        <Space height={s(10)} />
        <View style={styles.input}>{children}</View>
        <Space height={s(100)} />
      </ScrollView>
    </NeomorphView>
  )
}

const styles = StyleSheet.create({
  body: {
    color: gray,
    fontFamily: 'Montserrat-Regular',
    fontSize: Platform.OS === 'ios' ? s(15) : s(15),
    fontWeight: 'normal',
  },
  container: {
    alignSelf: 'center',
    height: '100%',
    paddingHorizontal: 15,
    paddingVertical: 20,
    width: '90%',
  },
  heading1: {
    color: gray,
    fontFamily: Platform.OS === 'ios' ? 'mont' : 'mont',
    fontSize: Platform.OS === 'ios' ? ms(25, 0.5) : ms(30, 0.3),
    fontWeight: 'bold',
    textShadowOffset: { width: 1, height: 1 },
    textShadowRadius: 1,
  },
  heading2: {
    color: gray,
    fontFamily: 'mont',
    fontSize: Platform.OS === 'ios' ? s(20) : s(20),
  },
  heading3: {
    color: gray,
    fontFamily: 'mont',
    fontSize: Platform.OS === 'ios' ? ms(18, 0.6) : ms(18, 0.6),
    textShadowOffset: { width: 1, height: 1 },
    textShadowRadius: 1,
  },
  heading4: {
    color: gray,
    fontFamily: 'mont',
    fontSize: Platform.OS === 'ios' ? ms(15, 0.8) : s(15),
    textShadowRadius: 1,
  },
  heading5: {
    color: gray,
    fontFamily: Platform.OS === 'ios' ? 'mont' : 'mont',
    fontSize: Platform.OS === 'ios' ? s(15) : s(15),
  },
  heading6: {
    color: gray,
    fontFamily: 'mont',
    fontSize: Platform.OS === 'ios' ? s(15) : s(15),
  },
  input: {
    alignSelf: 'center',
    width: W,
  },
})

export { MarkdownView }
