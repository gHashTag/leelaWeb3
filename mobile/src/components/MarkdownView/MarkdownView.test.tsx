import React from 'react'

import { View } from 'react-native'

import { render } from '@testing-library/react-native'
import renderer from 'react-test-renderer'

import { MarkdownView } from './MarkdownView'

describe('MarkdownView Component', () => {
  const mockMarkdown =
    '## Heading 2\n\nThis is some **bold** and *italic* text.\n\n- Item 1\n- Item 2\n- Item 3'
  const mockChildren = <View testID="mock-children" />

  it('matches snapshot', () => {
    const tree = renderer
      .create(<MarkdownView markdown={mockMarkdown} children={mockChildren} />)
      .toJSON()
    expect(tree).toMatchSnapshot()
  })

  it('renders correctly', () => {
    const { getByTestId, getByText } = render(
      <MarkdownView markdown={mockMarkdown} children={mockChildren} />,
    )

    const markdownHeading = getByText('Heading 2')
    const boldText = getByText('bold')
    const italicText = getByText('italic')
    const item1 = getByText('Item 1')
    const item2 = getByText('Item 2')
    const item3 = getByText('Item 3')
    const childrenView = getByTestId('mock-children')

    // Check if the markdown content renders
    expect(markdownHeading).toBeTruthy()
    expect(boldText).toBeTruthy()
    expect(italicText).toBeTruthy()
    expect(item1).toBeTruthy()
    expect(item2).toBeTruthy()
    expect(item3).toBeTruthy()

    // Check if the children view renders
    expect(childrenView).toBeTruthy()
  })
})
