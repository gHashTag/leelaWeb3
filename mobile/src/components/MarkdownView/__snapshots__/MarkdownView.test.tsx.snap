// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MarkdownView Component matches snapshot 1`] = `
<View
  style={
    {
      "backfaceVisibility": undefined,
      "borderRadius": 20,
      "bottom": undefined,
      "direction": undefined,
      "display": undefined,
      "end": undefined,
      "height": undefined,
      "left": undefined,
      "margin": undefined,
      "marginBottom": undefined,
      "marginEnd": undefined,
      "marginHorizontal": undefined,
      "marginLeft": undefined,
      "marginRight": undefined,
      "marginStart": undefined,
      "marginTop": undefined,
      "marginVertical": undefined,
      "opacity": undefined,
      "position": undefined,
      "right": undefined,
      "start": undefined,
      "top": undefined,
      "transform": undefined,
      "width": undefined,
      "zIndex": undefined,
    }
  }
>
  <View
    style={
      [
        {
          "backgroundColor": "#F5F7F8",
          "borderRadius": 20,
          "height": undefined,
          "shadowColor": "black",
          "shadowOffset": {
            "x": 5,
            "y": 5,
          },
          "shadowOpacity": 0.04970404046150326,
          "shadowRadius": 10,
          "width": undefined,
        },
        {
          "position": "absolute",
          "shadowOffset": {
            "height": 5,
            "width": 5,
          },
          "shadowRadius": 5,
        },
      ]
    }
  />
  <View
    style={
      [
        {
          "backgroundColor": "#F5F7F8",
          "borderRadius": 20,
          "height": undefined,
          "shadowColor": "white",
          "shadowOffset": {
            "x": -5,
            "y": -5,
          },
          "shadowOpacity": 0.8615387444286695,
          "shadowRadius": 10,
          "width": undefined,
        },
        {
          "position": "absolute",
          "shadowOffset": {
            "height": -5,
            "width": -5,
          },
          "shadowRadius": 5,
        },
      ]
    }
  />
  <View
    style={
      {
        "backgroundColor": "#F5F7F8",
        "borderRadius": 20,
        "height": undefined,
        "width": undefined,
      }
    }
  >
    <View>
      <RCTScrollView
        contentInsetAdjustmentBehavior="automatic"
        showsVerticalScrollIndicator={false}
        style={
          {
            "alignSelf": "center",
            "height": "100%",
            "paddingHorizontal": 15,
            "paddingVertical": 20,
            "width": "90%",
          }
        }
      >
        <View>
          <View
            style={{}}
          >
            <View
              style={
                {
                  "flexDirection": "row",
                }
              }
            >
              <Text
                style={{}}
              >
                <Text
                  style={
                    [
                      {
                        "color": "#808080",
                        "fontFamily": "mont",
                        "fontSize": 42.857142857142854,
                        "fontWeight": "normal",
                      },
                      {},
                    ]
                  }
                >
                  Heading 2
                </Text>
              </Text>
            </View>
            <View
              style={
                {
                  "alignItems": "flex-start",
                  "flexDirection": "row",
                  "flexWrap": "wrap",
                  "justifyContent": "flex-start",
                  "marginBottom": 10,
                  "marginTop": 10,
                  "width": "100%",
                }
              }
            >
              <Text
                style={{}}
              >
                <Text
                  style={
                    [
                      {
                        "color": "#808080",
                        "fontFamily": "Montserrat-Regular",
                        "fontSize": 32.14285714285714,
                        "fontWeight": "normal",
                      },
                      {},
                    ]
                  }
                >
                  This is some 
                </Text>
                <Text
                  style={
                    {
                      "fontWeight": "bold",
                    }
                  }
                >
                  <Text
                    style={
                      [
                        {
                          "color": "#808080",
                          "fontFamily": "Montserrat-Regular",
                          "fontSize": 32.14285714285714,
                          "fontWeight": "bold",
                        },
                        {},
                      ]
                    }
                  >
                    bold
                  </Text>
                </Text>
                <Text
                  style={
                    [
                      {
                        "color": "#808080",
                        "fontFamily": "Montserrat-Regular",
                        "fontSize": 32.14285714285714,
                        "fontWeight": "normal",
                      },
                      {},
                    ]
                  }
                >
                   and 
                </Text>
                <Text
                  style={
                    {
                      "fontStyle": "italic",
                    }
                  }
                >
                  <Text
                    style={
                      [
                        {
                          "color": "#808080",
                          "fontFamily": "Montserrat-Regular",
                          "fontSize": 32.14285714285714,
                          "fontStyle": "italic",
                          "fontWeight": "normal",
                        },
                        {},
                      ]
                    }
                  >
                    italic
                  </Text>
                </Text>
                <Text
                  style={
                    [
                      {
                        "color": "#808080",
                        "fontFamily": "Montserrat-Regular",
                        "fontSize": 32.14285714285714,
                        "fontWeight": "normal",
                      },
                      {},
                    ]
                  }
                >
                   text.
                </Text>
              </Text>
            </View>
            <View
              style={{}}
            >
              <View
                style={
                  {
                    "flexDirection": "row",
                    "justifyContent": "flex-start",
                  }
                }
              >
                <Text
                  accessible={false}
                  style={
                    [
                      {
                        "color": "#808080",
                        "fontFamily": "Montserrat-Regular",
                        "fontSize": 32.14285714285714,
                        "fontWeight": "normal",
                      },
                      {
                        "marginLeft": 10,
                        "marginRight": 10,
                      },
                    ]
                  }
                >
                  ·
                </Text>
                <View
                  style={
                    {
                      "flex": 1,
                    }
                  }
                >
                  <Text
                    style={{}}
                  >
                    <Text
                      style={
                        [
                          {
                            "color": "#808080",
                            "fontFamily": "Montserrat-Regular",
                            "fontSize": 32.14285714285714,
                            "fontWeight": "normal",
                          },
                          {},
                        ]
                      }
                    >
                      Item 1
                    </Text>
                  </Text>
                </View>
              </View>
              <View
                style={
                  {
                    "flexDirection": "row",
                    "justifyContent": "flex-start",
                  }
                }
              >
                <Text
                  accessible={false}
                  style={
                    [
                      {
                        "color": "#808080",
                        "fontFamily": "Montserrat-Regular",
                        "fontSize": 32.14285714285714,
                        "fontWeight": "normal",
                      },
                      {
                        "marginLeft": 10,
                        "marginRight": 10,
                      },
                    ]
                  }
                >
                  ·
                </Text>
                <View
                  style={
                    {
                      "flex": 1,
                    }
                  }
                >
                  <Text
                    style={{}}
                  >
                    <Text
                      style={
                        [
                          {
                            "color": "#808080",
                            "fontFamily": "Montserrat-Regular",
                            "fontSize": 32.14285714285714,
                            "fontWeight": "normal",
                          },
                          {},
                        ]
                      }
                    >
                      Item 2
                    </Text>
                  </Text>
                </View>
              </View>
              <View
                style={
                  {
                    "flexDirection": "row",
                    "justifyContent": "flex-start",
                  }
                }
              >
                <Text
                  accessible={false}
                  style={
                    [
                      {
                        "color": "#808080",
                        "fontFamily": "Montserrat-Regular",
                        "fontSize": 32.14285714285714,
                        "fontWeight": "normal",
                      },
                      {
                        "marginLeft": 10,
                        "marginRight": 10,
                      },
                    ]
                  }
                >
                  ·
                </Text>
                <View
                  style={
                    {
                      "flex": 1,
                    }
                  }
                >
                  <Text
                    style={{}}
                  >
                    <Text
                      style={
                        [
                          {
                            "color": "#808080",
                            "fontFamily": "Montserrat-Regular",
                            "fontSize": 32.14285714285714,
                            "fontWeight": "normal",
                          },
                          {},
                        ]
                      }
                    >
                      Item 3
                    </Text>
                  </Text>
                </View>
              </View>
            </View>
          </View>
          <View
            style={
              {
                "height": 45.91836734693877,
                "width": 0,
              }
            }
            testID="space-component"
          />
          <View
            style={
              {
                "alignSelf": "center",
                "width": 750,
              }
            }
          >
            <View
              testID="mock-children"
            />
          </View>
          <View
            style={
              {
                "height": 459.1836734693877,
                "width": 0,
              }
            }
            testID="space-component"
          />
        </View>
      </RCTScrollView>
    </View>
  </View>
</View>
`;
